app/celer-g4/LogHandlers.cc

```diff
@@ -16,11 +16,34 @@ namespace celeritas
 {
 namespace app
 {
+namespace
+{
+//---------------------------------------------------------------------------//
+//! Finalize the message, including
+void write_msg(std::ostringstream&& os,
+               LogProvenance const& prov,
+               LogLevel const& lev,
+               std::string const& msg)
+{
+    // Write main message
+    StreamLogHandler{os}(prov, lev, msg);
+
+    // Lock after building the message while writing
+    static std::mutex log_mutex;
+    std::lock_guard scoped_lock{log_mutex};
```

Comment:
Do we get here for every 'potential' message or is the code short-circuited early for message that are to be ignored?

Reply:
This only applies if the log level is "visible": the only way it won't write now is to not deliberately not print.